arches/app/utils/geo_utils.py

```diff
@@ -94,3 +99,142 @@ def convert_geos_geom_collection_to_feature_collection(self, geometry):
+        and returns a FeatureCollection of the original features with the buffer distance added.
+
+        """
+
+        buffered_features = []
+
+        for feature in feature_collection["features"]:
+            geom = GEOSGeometry(json.dumps(feature["geometry"]))
+
+            geom.transform(settings.ANALYSIS_COORDINATE_SYSTEM_SRID)
+            buffered_geom = geom.buffer(buffer_distance)
+            buffered_geom.transform(4326)
+
+            buffered_feature = {
+                "type": "Feature",
+                "id": str(uuid.uuid4()),
+                "properties": feature.get("properties", {}),
+                "geometry": json.loads(buffered_geom.geojson),
+            }
+            buffered_features.append(buffered_feature)
+
+        return {
+            "type": "FeatureCollection",
+            "features": buffered_features,
+        }
+
+    def get_intersection_and_difference_of_feature_collections(
+        self, feature_collection_1, feature_collection_2
+    ):
+        def _build_feature_collection(geometries):
+            features = []
+
+            if geometries is None or geometries.empty:
+                return {"type": "FeatureCollection", "features": []}
+
+            if geometries.geom_type in [
+                "GeometryCollection",
+                "MultiPolygon",
+                "MultiLineString",
+                "MultiPoint",
+            ]:
+                for geometry in geometries:
+                    if not geometry.empty:
+                        features.append(
+                            {
+                                "type": "Feature",
+                                "id": str(uuid.uuid4()),
+                                "properties": {},
+                                "geometry": json.loads(geometry.geojson),
+                            }
+                        )
+            else:
+                features.append(
+                    {
+                        "type": "Feature",
+                        "id": str(uuid.uuid4()),
+                        "properties": {},
+                        "geometry": json.loads(geometries.geojson),
+                    }
+                )
+
+            return {"type": "FeatureCollection", "features": features}
+
+        feature_collection_1_geometries = [
+            GEOSGeometry(json.dumps(feature["geometry"]))
+            for feature in feature_collection_1["features"]
+        ]
+        feature_collection_1_geometries_union = GeometryCollection(
+            *feature_collection_1_geometries
+        ).unary_union
+
+        feature_collection_2_geometries = [
+            GEOSGeometry(json.dumps(feature["geometry"]))
+            for feature in feature_collection_2["features"]
+        ]
+        feature_collection_2_geometries_union = GeometryCollection(
+            *feature_collection_2_geometries
+        ).unary_union
+
+        return {
```

Comment:
i'm thinking it might be better to just return a single feature collection here and use properties on the features rather than keys at the top level to distinguish "types" of features.  this would then return actual geojson rather than geojson nested inside an object which seems more usable